src/mmw/apps/user/hydroshare.py

```diff
@@ -12,6 +13,7 @@
 from django.conf import settings
 
 from apps.user.models import HydroShareToken
+from apps.modeling.tasks import to_gms_file
```

Comment:
Importing this here feels like it couples this too much with modeling. Some of this will be refactored in #2584, but I think it would be a better idea to use `to_gms_file` in `views.py`, and use it to construct a `{name: 'scenario_xyz.gms', contents: '...'}` object that is added to the `files` array given to `hs.add_files`. Does this make sense?

Reply:
Yes, but it won't be as clean as described.`to_gms_file` returns a file, and shouldn't be passed as `contents` to `hs.add_files` which will try to rewrite it via `StringIO`.

Reply:
Ah right. Very well, we'll tackle that later.